docs/docs/subscription.fsx

```diff
@@ -23,61 +23,141 @@ We can setup forwarding of those events to be processed by our `update` function
 Let's define our `Model` and `Msg` types. `Model` will hold the current state and `Msg` will tell us the nature of the change that we need to apply to the current state.
 *)
 
+open Elmish
+open Fable.Core
 open System
 
 
-type Model =
-    {
-        current : DateTime
-    }
+module Example1 =
 
-type Msg =
-    | Tick of DateTime
+    type Model =
+        {
+            current : DateTime
+        }
+
+    type Msg =
+        | Tick of DateTime
 
 (**
 This time we'll define the "simple" version of `init` and `update` functions, that don't produce commands:
 *)
 
-let init () =
-    {
-        current = DateTime.Now
-    }
-
-let update msg model =
-    match msg with
-    | Tick current ->
-        { model with
-            current = current
+    let init () =
+        {
+            current = DateTime.MinValue
         }
 
+    let update msg model =
+        match msg with
+        | Tick current ->
+            { model with
+                current = current
+            }
+
 (**
 
-Note that "simple" is not a requirement and is just a matter of convenience for the purpose of the example!
+Note that "simple" is not a requirement and is just a matter of convenience for the example!
 
 Now lets define our timer subscription:
 
 *)
-open Elmish
-open Fable.Core
+    let timer onTick =
+        let start dispatch =
+            let intervalId = 
+                JS.setInterval
+                    (fun _ -> dispatch (onTick DateTime.Now))
+                    1000
+            { new IDisposable with
+                member _.Dispose() = JS.clearInterval intervalId }
+        start
+
+    let subscribe model =
+        [ ["timer"], timer Tick ]
 
-let timer initial =
-    let sub dispatch =
-        JS.setInterval
-            (fun _ ->
-                dispatch (Tick DateTime.Now)
-            )
-            1000
-            |> ignore
-    Cmd.ofSub sub
+    Program.mkSimple init update (fun model _ -> printf "%A\n" model)
+    |> Program.withSubscription subscribe
+    |> Program.run
+
+
+(**
+`subscribe` answers the question: "Which subscriptions should be running?" It has the current program state, `model`, to use for decisions.
+When the model changes, `subscribe` is called. Elmish then starts or stops subscriptions to match what should be running.
 
-Program.mkSimple init update (fun model _ -> printf "%A\n" model)
-|> Program.withSubscription timer
-|> Program.run
+A subscription has an ID, `["timer"]` here, and a start function. You're probably wondering why the ID is a list.
+It allows you to include dependencies. Later we will use this to make the timer change intervals.
 
+Here, the timer subscription is always returned, so it will stay running as long as the program is running.
+
+Let's look at an example where the timer can be turned off. First we add the field `enabled`.
+*)
+
+module Example2 =
```

Comment:
@kspeakman If you want to avoid the user seeing `module Example2 =` in the ouput.

You can use `(*** hide ***)` and `(*** show ***)` to hide portion of the F# code.

*Note: `(*** hide ***) will hide the code up to the next `(**` comments.*

Example:

```fs
(*** hide ***)
let a = 42 // This will not be in the output 

(*** show ***)
let b = 42 // This will be in the ouput
```


```fs
(*** hide ***)
let a = 42 // This will not be in the output 

(**
This text will be in the output and the next snippet too
*)
let b = 42 // This will be in the ouput
```

Reply:
Thanks for the feedback. I made some changes based on it. I tried to make the module names match the example's purpose. For one-off notes, I hid the module name to make it appear as if it was in the previously module.